ProcessLib/TwoPhaseFlowWithPrho/TwoPhaseFlowWithPrhoMaterialProperties.h

```diff
@@ -0,0 +1,278 @@
+    bool computeConstitutiveRelation(
+        double const t,
+        ProcessLib::SpatialPosition const& x_position,
+        double const PG,
+        double const X,
+        double& Sw,
+        double& X_m,
+        double& dsw_dpg,
+        double& dsw_dX,
+        double& dxm_dpg,
+        double& dxm_dX);
+
+protected:
+    /// A flag to indicate whether the reference member, _material_ids,
+    /// is not assigned.
+    const bool _has_material_ids;
+
+    std::unique_ptr<MaterialLib::Fluid::FluidProperty> _liquid_density;
+    std::unique_ptr<MaterialLib::Fluid::FluidProperty> _viscosity;
+    std::unique_ptr<MaterialLib::Fluid::FluidProperty> _gas_density;
+    std::unique_ptr<MaterialLib::Fluid::FluidProperty> _gas_viscosity;
+
+    /** Use two phase models for different material zones.
+    *  Material IDs must be given as mesh element properties.
+    */
+    MeshLib::PropertyVector<int> const& _material_ids;
+
+    int _current_material_id = 0;
+    std::vector<Eigen::MatrixXd> _intrinsic_permeability_models;
+    std::vector<std::unique_ptr<MaterialLib::PorousMedium::Porosity>>
+        _porosity_models;
+    std::vector<std::unique_ptr<MaterialLib::PorousMedium::Storage>>
+        _storage_models;
+    std::vector<
+        std::unique_ptr<MaterialLib::PorousMedium::CapillaryPressureSaturation>>
+        _capillary_pressure_models;
+    std::vector<
+        std::unique_ptr<MaterialLib::PorousMedium::RelativePermeability>>
+        _relative_permeability_models;
+
+private:
+    /// Calculates the 18x1 residual vector.
+    void calculateResidual(double const PG, double const X, double Sw,
+                           double X_m, ResidualVector& res);
+
+    /// Calculates the 18x18 Jacobian.
+    void calculateJacobian(double const t, ProcessLib::SpatialPosition const& x,
+                           double const PG, double const X, JacobianMatrix& Jac,
+                           double Sw, double X_m);
+
+    /**
+    * Complementary condition 1
+    * for calculating molar fraction of light component in the liquid phase
+    */
+    double Calc_equili_rho_m(double const PG, double const Sw,
+                             double const rho_h2_wet) const
+    {
+        double const rho_equili_h2_wet = PG * Hen * molar_mass_h2;
+        return std::min(1 - Sw, rho_equili_h2_wet - rho_h2_wet);
+    }
+    /**
+    * Complementary condition 2
+    * for calculating the saturation
+    */
+    double Calc_Saturation(double /*PL*/, double X, double Sw,
+                           double rho_h2_wet, double rho_h2_nonwet,
+                           double /*T*/) const
+    {
+        return X - (Sw * rho_h2_wet + (1 - Sw) * rho_h2_nonwet);
+    }
+    /**
+    * Calculate the derivatives using the analytical way
+    */
+    double Calculate_dSwdP(double PL, double S, double rho_h2_wet,
+                           double const T = 303.15) const
+    {
+        const double PG = PL +
```

Comment:
Sometimes Pg is used, sometimes p_g, sometimes PG. 
Please use p_g consistently.